x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/dataframe/transforms/pivot/GroupConfigTests.java

```diff
@@ -27,6 +28,9 @@
 
 public class GroupConfigTests extends AbstractSerializingTestCase<GroupConfig> {
 
+    // array of illegal characters, see {@link AggregatorFactories#VALID_AGG_NAME}
+    private static final char[] ILLEGAL_FIELD_NAME_CHARACTERS = {'^', '[', ']', '>'};
```

Comment:
I think the `^` in `AggregatorFactories#VALID_AGG_NAME` is the regular expression "complement of" operator rather than a disallowed character.  So it's not necessary to ban it for consistency with aggregations.